compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -296,7 +296,7 @@ class ExpressionCodegen(
     }
 
     private fun writeParameterInLocalVariableTable(startLabel: Label, endLabel: Label) {
-        if (irFunction.origin == IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER) return
+        if (!irFunction.isInline && irFunction.origin == IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER) return
```

Comment:
Could you explain a bit (because in my head all variable declarations should be inlined from original function to default one)?

Reply:
It's mostly about matching the behavior of the JVM BE, which normally does not produce LVT entries for the synthetic `$default` methods. For inline methods these entries are required for debugging, though. We just hide the `$mask` and `$handler` arguments in this case.

I don't think there would be anything wrong with always producing LVT entries, except that we have several bytecode text tests which check that we don't.

Reply:
In current pr added generation for LVT in case of inline $default function, but same variables should be also inlined from original function. Looks like here would be a duplication

Reply:
There isn't, since these variables are not shifted and so we don't generate LVT entries for them. This may be specific to `$default` methods. I don't know enough about the inliner to say for sure. :/

Reply:
@sfs Could you look a bit deeper in $default inline function compilation?
During compilation of $default inline function call to original one is inlined into it.
And all variables and parameters from original one should be inlined into $default one. It's strange that `p`  declaration was not inlined.
There are at least one more inconsistency: all locals variables are renamed with $iv suffix, e.g. :+1: 

```
inline fun test(p: String = "OK"): String {
    var x = 1
    return p
}

fun box() : String {
    return test()
}
//  public static synthetic test$default(Ljava/lang/String;ILjava/lang/Object;)Ljava/lang/String;
//    LOCALVARIABLE $i$f$test I L3 L5 3
//    LOCALVARIABLE x$iv I L4 L5 4
// old backend: x
//    LOCALVARIABLE p Ljava/lang/String; L0 L6 0


//  public final static box()Ljava/lang/String;
//    LOCALVARIABLE $i$f$test I L4 L6 3
//    LOCALVARIABLE x$iv$iv I L5 L6 4
// old backend: x$iv
//    LOCALVARIABLE p$iv Ljava/lang/String; L2 L7 0

```
But in old backend they are kept with original names (see InlineCodegenForDefaultBody)
cc @pyos

Reply:
I'll look into it some more, especially the suffix for local variables looks like it might become a problem (or does the debugger ignore it?).

What I already know is that the variables are inlined into the $default method, but we only create LVT entries of the form `p$ip` for variables with kind `SHIFT` (see org/jetbrains/kotlin/codegen/inline/LocalVarRemapper.kt:93). For this particular call the arguments end up with kind `REMAPPED` and that's why the LVT entries are missing.

This seems different from what the JVM BE is doing. I'll step through it and see what's going on.

Reply:
> or does the debugger ignore it?

Maybe @yanex can help